fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -148,7 +148,9 @@ public void processTaskRun(TaskRunContext context) throws Exception {
 
     private void updateMeta(ExecPlan execPlan) {
         // update the meta if succeed
-        database.writeLock();
+        if (!database.writeLockAndCheckExist()) {
+            throw new DmlException("update meta failed. database:" + database.getFullName() + " not exist");
+        }
```

Comment:
The table or partition may be dropped too. So you should check the existence of table/partition before update.